# Analysing Profitability of Apps on Apple and Google stores

The aim of this project is to find apps that are potentially profitable in Apple and Google Play stores. The insights learned from this analysis may be helpful to people seeking to understand the mobile app ecosystem better, such as the profitable domains or functions.

As a data analyst for Company X that builds mobile iOS and Android apps, my goal is to find the types of apps that attract users, and in turn, help X find profitable channels, especially for companies looking to place in-app ads.

[This](https://www.kaggle.com/lava18/google-play-store-apps) is the data set containing data about approximately 10,000 Android apps from Google Play; the data was collected in August 2018. 
In addition, [this](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps) is the link of the iOS apps from the App Store, collected in July 2017.

### Data Exploration

Since the entire dataset of apps available in App Store and Play store go over millions, the sample dataset as provided in the links above will be used instead.

In [33]:
from csv import reader
open_file = open('googleplaystore.csv')
read_file = reader(open_file)
google = list(read_file)
google_header = google[0]
google = google[1:]
            
open_file = open('AppleStore.csv')
read_file = reader(open_file)
apple = list(read_file)
apple_header = apple[0]
apple = apple[1:]

In [34]:
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n')

    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))

In [35]:
explore_data(google,0,5,True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


Number of rows: 10841
Number of columns: 13


There are 10841 Android apps in this dataset, and 13 columns.

In [36]:
print(google_header)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


In [37]:
explore_data(apple, 0,5,True)

['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


['284035177', 'Pandora - Music & Radio', '130242560', 'USD', '0.0', '1126879', '3594', '4.0', '4.5', '8.4.1', '12+', 'Music', '37', '4', '1', '1']


Number of rows: 7197
Number of columns: 16


Apple store has 7197 apps and 16 columns:

In [38]:
print(apple_header)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


For more information on the columns, check this [documentation](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps/home).

### Data Cleaning: Deleting wrong data

In [39]:
print(google[10472])

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


In [40]:
print(google_header)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


In [41]:
print(google[0])

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


In [42]:
print(len(google))
del google[10472]
print(len(google))

10841
10840


### Data Cleaning: Removing Duplicate Entries

### Part One

Exploring duplicate entries from datasets

In [43]:
duplicate_apps = []
unique_apps = []

for app in google:
    name = app[0]
    if name in unique_apps:
        duplicate_apps.append(name)
    else:
        unique_apps.append(name)

print('Number of duplicate apps:', len(duplicate_apps))
print('\n')
print('Examples of duplicate apps:', duplicate_apps[:15])

Number of duplicate apps: 1181


Examples of duplicate apps: ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic', 'Insightly CRM', 'QuickBooks Accounting: Invoicing & Expenses', 'HipChat - Chat Built for Teams', 'Xero Accounting Software']


The total of duplicates is 1181 and above are some examples of them. We need to delete the duplicates by examining the number of reviews of each. Since higher number of reviews imply that that particular entry is more recent, we will retain that entry and delete the rest.

To do that, we will (1) create a dictionary containing a unique app name and the highest review and (2) create a new dataset with the dictionary, ensuring that there will be only one entry per app, and the entry has the highest number of reviews amongst its duplicates.

### Part Two

Create a dictionary.

In [44]:
reviews_max = {}

for app in google:
    name = app[0]
    n_reviews = float(app[3])
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
    elif name not in reviews_max:
        reviews_max[name] = n_reviews        

To check whether the function will generate accurate dictionary, we can check the length of it.

In [45]:
#This is the actual length:
len(reviews_max)

9659

In [46]:
#This is the expected length:
len(google) - 1181

9659

In [47]:
android_clean = []
already_added = []

for app in google:
    name = app[0]
    n_reviews = float(app[3])
    
    if (reviews_max[name] == n_reviews) and (name not in already_added):
        android_clean.append(app)
        already_added.append(name)

In [48]:
explore_data(android_clean, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9659
Number of columns: 13


There are 9,659 rows as expected.

### Data Cleaning: Removing Non-English Apps

### Part One

In [49]:
def eng_app(string):
    for char in string:
        if ord(char) > 127:
            return False
    return True

Testing eng_app function.

In [50]:
print(eng_app('Instagram'))
print(eng_app('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(eng_app('Docs To Go™ Free Office Suite'))
print(eng_app('Instachat 😜'))

True
False
False
False


The function is not able to detect English apps with special characters. The function is refined so that it only excludes apps with more than 3 non-english characters.

### Part Two

In [51]:
def is_eng(string):
    ascii_count = 0

    for char in string:
        if ord(char) > 127:
            ascii_count += 1
    
    if ascii_count > 3:
        return False
    else:
        return True

In [52]:
print(is_eng('Instagram'))
print(is_eng('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(is_eng('Docs To Go™ Free Office Suite'))
print(is_eng('Instachat 😜'))

True
False
True
True


Use is_eng function to create a list of filtered non-English apps for both datasets.

In [53]:
android_english = []
apple_english = []

for app in android_clean:
    name = app[0]
    if is_eng(name):
        android_english.append(app)
        
for app in apple:
    name = app[1]
    if is_eng(name):
        apple_english.append(app)

In [54]:
explore_data(android_english, 0, 3, True)
print('\n')
explore_data(apple_english, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9614
Number of columns: 13


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 

There are 9614 Android apps and 6183 iOS apps.

### Data Cleaning: Isolating Free Apps

In [55]:
android_free = []
apple_free = []

for app in android_english:
    price = app[7]
        
    if price == '0':
        android_free.append(app)
    
for app in apple_english:
    price = app[4]
        
    if price == '0.0':
        apple_free.append(app)
        
print(len(android_free))
print(len(apple_free))

        

8864
3222


We are left with 8864 Android and 3222 iOS apps.

### Data Analysis: Analysing Apps by Genre

### Part One

#### Validation Strategy

1. Build a minimal Android version of the app, then add it to Google Play.
2. If the app has a good response from users, we develop it further.
3. If the app is profitable after 6 months, we build an iOS version of the app and add it to the App Store.

We start by analysing the most common genres for each market. We build a frequency table for a few columns in the datasets.
The end goal is to find app profiles that are successful in both markets.
We start with exploring the columns prime_genre for App Store, and Genres and Category for Play Store.

### Part Two

Below is a function that takes in a dataset in the form of a list of lists, and an integer index.

The first function generates a frequency table.
The second function displays the frequency table in a descending order so we can immediately see the highest frequency genres.

In [56]:
def freq_table(dataset, index):
    table = {}
    total = 0
    
    for row in dataset:
        total += 1
        value = row[index]
        if value in table:
            table[value] += 1
        else:
            table[value] = 1
    
    table_per = {}
    for key in table:
        percent = (table[key]/total) * 100
        table_per[key] = percent
        
    return table_per

def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)

    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])
    

### Part Three

In [57]:
display_table(apple_free, -5)

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


The top 3 most common genres for App Store include Games (58%), Entertainment (7.8%), and Photo and Video (4.9%). The runner-ups include Education (3.6%), Social Networking (3.2%) and Shopping (2.6%).

In [58]:
display_table(android_free, 1)

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

The top 3 most common genres for Play Store include Family (19%), Game (9.7%) and Tools (8.4%). The runner-ups include Business (4.5%), Lifestyle (3.9%) and Productivity (3.8%).

Although the overlapping genre across the two markets is mainly Gaming, we can see that the market share in App Store holds more than half of the total.
Most apps in Play Store are more functional i.e. productivity, business and tools.

Looking at the current Play Store however, the 'Family' genre has been removed. In addition, the 'Games' genre has an entire list of subcategories of its own, which proves its dominance in the market.

### Data Analysis: Most Popular Apps within Genres by Market

Another way to find the most popular genres is to take the average number of installs for each app genre.

### App Store

Because the number of installs is not directly available on the dataset, we will create a function to extract that information.
To do so, we need to:
1. Isolate the apps of each genre.
2. Sum up the user ratings for the apps of that genre.
3. Divide the sum by the number of apps belonging to that genre.

In [61]:
apple_genre = freq_table(apple_free, -5)

for genre in apple_genre:
    total = 0
    len_genre = 0
    
    for data in apple_free:
        genre_app = data[-5]
        if genre_app == genre:
            n_ratings = float(data[5])
            total += n_ratings
            len_genre += 1
    avg_n_ratings = total / len_genre
    print(genre, ':', avg_n_ratings)
        

Health & Fitness : 23298.015384615384
Finance : 31467.944444444445
Medical : 612.0
Catalogs : 4004.0
Utilities : 18684.456790123455
Food & Drink : 33333.92307692308
Weather : 52279.892857142855
Lifestyle : 16485.764705882353
Business : 7491.117647058823
Sports : 23008.898550724636
Games : 22788.6696905016
Reference : 74942.11111111111
Navigation : 86090.33333333333
Entertainment : 14029.830708661417
Productivity : 21028.410714285714
Travel : 28243.8
Social Networking : 71548.34905660378
Book : 39758.5
News : 21248.023255813954
Photo & Video : 28441.54375
Music : 57326.530303030304
Shopping : 26919.690476190477
Education : 7003.983050847458


We can observe that the 'Navigation' genre has the highest reviews.

In [63]:
for app in apple_free:
    if app[-5] == 'Navigation':
        print(app[1],':', app[5])

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Google Maps - Navigation & Transit : 154911
Geocaching® : 12811
CoPilot GPS – Car Navigation & Offline Maps : 3582
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5


The Navigation genre is mainly dominated by Google and Waze that have almost half the market together. The succeeding apps have a wide gap with these two genre leaders.

In [64]:
for app in apple_free:
    if app[-5] == 'Reference':
        print(app[1], ":", app[5])

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Google Translate : 26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
Merriam-Webster Dictionary : 16849
Night Sky : 12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
WWDC : 762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
VPN Express : 14
Real Bike Traffic Rider Virtual Reality Glasses : 8
教えて!goo : 0
Jishokun-Japanese English Dictionary & Translator : 0


The same pattern is observed in the References genre where Bible and Dictionary.com together take more than half of the market.
This looks like an attractive genre for digitalising any book or paper reference. In addition, since App Store is dominantly fun apps such as Gaming ones, there's a potential in merging this model of app with Games i.e. online crosswords and quizzes.

### Play Store

Play Store dataset has the number of installs available, however, they are not precise enough i.e. they are open-ended values.
Since the values are in strings i.e. '1000+', we need to convert this into float value type in order to use it and compute for average users.

In [65]:
display_table(android_free, 5)

1,000,000+ : 15.726534296028879
100,000+ : 11.552346570397113
10,000,000+ : 10.548285198555957
10,000+ : 10.198555956678701
1,000+ : 8.393501805054152
100+ : 6.915613718411552
5,000,000+ : 6.825361010830325
500,000+ : 5.561823104693141
50,000+ : 4.7721119133574
5,000+ : 4.512635379061372
10+ : 3.5424187725631766
500+ : 3.2490974729241873
50,000,000+ : 2.3014440433213
100,000,000+ : 2.1322202166064983
50+ : 1.917870036101083
5+ : 0.78971119133574
1+ : 0.5076714801444043
500,000,000+ : 0.2707581227436823
1,000,000,000+ : 0.22563176895306858
0+ : 0.04512635379061372
0 : 0.01128158844765343


In [75]:
android_cat = freq_table(android_free, 1)

for category in android_cat:
    total = 0
    len_cat = 0
    
    for app in android_free:
        category_app = app[1]
        if category_app == category:
            n_installs = app[5]
            n_installs = n_installs.replace(',','')
            n_installs = n_installs.replace('+','')
            total += float(n_installs)
            len_cat += 1
    avg_n_installs = total / len_cat
    print(category,':', avg_n_installs)

VIDEO_PLAYERS : 24727872.452830188
COMMUNICATION : 38456119.167247385
PHOTOGRAPHY : 17840110.40229885
MAPS_AND_NAVIGATION : 4056941.7741935486
GAME : 15588015.603248259
TRAVEL_AND_LOCAL : 13984077.710144928
ENTERTAINMENT : 11640705.88235294
NEWS_AND_MAGAZINES : 9549178.467741935
EVENTS : 253542.22222222222
PARENTING : 542603.6206896552
HEALTH_AND_FITNESS : 4188821.9853479853
ART_AND_DESIGN : 1986335.0877192982
FAMILY : 3695641.8198090694
AUTO_AND_VEHICLES : 647317.8170731707
PERSONALIZATION : 5201482.6122448975
BOOKS_AND_REFERENCE : 8767811.894736841
BEAUTY : 513151.88679245283
COMICS : 817657.2727272727
DATING : 854028.8303030303
MEDICAL : 120550.61980830671
HOUSE_AND_HOME : 1331540.5616438356
SPORTS : 3638640.1428571427
LIBRARIES_AND_DEMO : 638503.734939759
SOCIAL : 23253652.127118643
FOOD_AND_DRINK : 1924897.7363636363
EDUCATION : 1833495.145631068
LIFESTYLE : 1437816.2687861272
FINANCE : 1387692.475609756
TOOLS : 10801391.298666667
SHOPPING : 7036877.311557789
PRODUCTIVITY : 167873

Communication apps have the highest install count. 

In [78]:
for app in android_free:
    if app[1] == 'COMMUNICATION' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

WhatsApp Messenger : 1,000,000,000+
imo beta free calls and text : 100,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
Viber Mess

Looking into Communication apps further, the category is dominated by three apps-- Whatsapp, Facebook Messenger, Google Chrome, Gmail and Hangouts, each with over 1 billion downloads.

Just like the Navigation category in App Store, the Communication category in Play Store are dominated by a few giants, who are difficult to compete against.


In [82]:
for app in android_free:
    if app[1] == 'VIDEO_PLAYERS' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

YouTube : 1,000,000,000+
Motorola Gallery : 100,000,000+
VLC for Android : 100,000,000+
Google Play Movies & TV : 1,000,000,000+
MX Player : 500,000,000+
Dubsmash : 100,000,000+
VivaVideo - Video Editor & Photo Movie : 100,000,000+
VideoShow-Video Editor, Video Maker, Beauty Camera : 100,000,000+
Motorola FM Radio : 100,000,000+


The same trend goes for the Video players category where Youtube dominates the rest. Looking at the convenience and amount of content and influence Youtube has, it does not look ideal that it will be replaced by other video playing apps. The same goes for Photography, which is dominated by Google Photos.

In [83]:
for app in android_free:
    if app[1] == 'PHOTOGRAPHY' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

B612 - Beauty & Filter Camera : 100,000,000+
YouCam Makeup - Magic Selfie Makeovers : 100,000,000+
Sweet Selfie - selfie camera, beauty cam, photo edit : 100,000,000+
Google Photos : 1,000,000,000+
Retrica : 100,000,000+
Photo Editor Pro : 100,000,000+
BeautyPlus - Easy Photo Editor & Selfie Camera : 100,000,000+
PicsArt Photo Studio: Collage Maker & Pic Editor : 100,000,000+
Photo Collage Editor : 100,000,000+
Z Camera - Photo Editor, Beauty Selfie, Collage : 100,000,000+
PhotoGrid: Video & Pic Collage Maker, Photo Editor : 100,000,000+
Candy Camera - selfie, beauty camera, photo editor : 100,000,000+
YouCam Perfect - Selfie Photo Editor : 100,000,000+
Camera360: Selfie Photo Editor with Funny Sticker : 100,000,000+
S Photo Editor - Collage Maker , Photo Collage : 100,000,000+
AR effect : 100,000,000+
Cymera Camera- Photo Editor, Filter,Collage,Layout : 100,000,000+
LINE Camera - Photo editor : 100,000,000+
Photo Editor Collage Maker Pro : 100,000,000+


In [86]:
for app in android_free:
    if app[1] == 'ENTERTAINMENT' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Hotstar : 100,000,000+
Talking Angela : 100,000,000+
IMDb Movies & TV : 100,000,000+
Talking Ben the Dog : 100,000,000+
Netflix : 100,000,000+


The Entertainment genre has a few players without a dominating app. If an app with unique value proposition can be built, this is a fairly attractive category to enter.

In [88]:
for app in android_free:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000+'
                                            or app[5] == '5,000,000+'
                                            or app[5] == '10,000,000+'
                                            or app[5] == '50,000,000+'):
        print(app[0], ':', app[5])

Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
AlReader -any text book reader : 5,000,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
ReadEra – free ebook reader : 1,000,000+
Ebook Reader : 5,000,000+
Read books online : 5,000,000+
eBoox: book reader fb2 epub zip : 1,000,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
Moon+ Reader : 10,000,000+
English-Myanmar Dictionary : 1,000,000+
Golden Dictionary (EN-AR) : 1,000,000+
All Language Translator Free : 1,000,000+
Aldiko Book Reader : 10,000,000+
Dictionary - WordWeb : 5,000,000+
50000 Free eBooks & Free AudioBooks : 5,000,000+
Al-Quran (Free) : 10,000,000+
Al Quran Indonesia : 10,000,000+
Al'Quran Bahasa Indonesia : 10,000,000+
Al Quran Al karim : 1,000,000+
Al Quran : EAlim - Translations & MP3 Offline : 5,000,000+
Koran Read &MP3 30 Juz Offline : 1,000,000+
H

Books and Reference is another industry with no few dominating giants. A lot of the apps are ebook or file readers and reference materials.

### Conclusion

Combining insights from App Store and Play Store, I recommend building an app that takes information from books or reference materials, and either allows users to read it electronically or transform the information into interesting content for entertainment such as audiobooks, podcast and the like. Other interesting app recommendation can transform content through gamification, perhaps to remember important points in a reference material better. These apps are the overlap of the few genres with dominating giants such as Gaming, Books and References and Entertainment.